File: Frontend/src/app/models/problem.ts
export interface IProblem {
	problemId: number
	title: string
	description: string
	solvedBy: number
	difficulty: string
	releaseDate: string
	tags: Array<ITag>
	isSolved: boolean
	result: IAnswer
}

export interface ITag {
	tagId: number
	tagName: string
}

export interface IAnswer {
	answer: string
	solvedDate: string
}

File: Backend/Backend/Models/Problem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using Backend.Models.Joins;

namespace Backend.Models
{
    public class Problem
    {
        public int ProblemId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Answer { get; set; }
        public int SolvedBy { get; set; }
        public int Difficulty { get; set; }
        public DateTime ReleaseDate { get; set; }
        public List<ProblemTag> ProblemTags { get; set; }
        public List<ProblemUser> ProblemUsers { get; set; }
        public List<Post> Posts { get; set; }
    }
}
